File: go/enums.go
package callisto

// OtpType is the character set used to generate an OTP code.
type OtpType string

const (
	OtpTypeDigit        OtpType = "digit"
	OtpTypeAlpha        OtpType = "alpha"
	OtpTypeAlphanumeric OtpType = "alphanumeric"
)

// OtpProvider is the delivery channel for an OTP.
type OtpProvider string

const (
	OtpProviderSMS      OtpProvider = "sms"
	OtpProviderWhatsApp OtpProvider = "whatsapp"
)

// WhatsAppMediaType is the type of media sent in a WhatsApp media message.
type WhatsAppMediaType string

const (
	WhatsAppMediaTypeImage    WhatsAppMediaType = "image"
	WhatsAppMediaTypeVideo    WhatsAppMediaType = "video"
	WhatsAppMediaTypeDocument WhatsAppMediaType = "document"
	WhatsAppMediaTypeAudio    WhatsAppMediaType = "audio"
)

// MessageStatus is the delivery status of an SMS or WhatsApp message.
type MessageStatus string

const (
	MessageStatusPending   MessageStatus = "pending"
	MessageStatusSent      MessageStatus = "sent"
	MessageStatusDelivered MessageStatus = "delivered"
	MessageStatusFailed    MessageStatus = "failed"
)

// OtpStatus is the lifecycle status of an OTP.
type OtpStatus string

const (
	OtpStatusPending  OtpStatus = "pending"
	OtpStatusVerified OtpStatus = "verified"
	OtpStatusExpired  OtpStatus = "expired"
	OtpStatusFailed   OtpStatus = "failed"
)

File: ruby/lib/callisto/enums.rb
# frozen_string_literal: true

module Callisto
  # Status of an SMS/WhatsApp message.
  module MessageStatus
    PENDING = "pending"
    SENT = "sent"
    DELIVERED = "delivered"
    FAILED = "failed"
  end

  # Status of an OTP.
  module OtpStatus
    PENDING = "pending"
    VERIFIED = "verified"
    EXPIRED = "expired"
    FAILED = "failed"
  end

  # Character set used when generating an OTP code.
  module OtpType
    DIGIT = "digit"
    ALPHA = "alpha"
    ALPHANUMERIC = "alphanumeric"
  end

  # Delivery channel for an OTP.
  module OtpProvider
    SMS = "sms"
    WHATSAPP = "whatsapp"
  end

  # WhatsApp media message type.
  module WhatsAppMediaType
    IMAGE = "image"
    VIDEO = "video"
    DOCUMENT = "document"
    AUDIO = "audio"
  end
end
